### PR TITLE
👋  `App.tsx`

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR removes `App.tsx` and its associated workings from DCR

## Why?
There are [two](https://docs.google.com/document/d/1gHOzvFZxrAO2mvNPKg4UmMAYZm7Jk1NutZPVLY5UxPY/edit#heading=h.1y29fcc1ygqg) [documents](https://docs.google.com/document/d/1kSrZZrJ8hSw6bek5hGa9onE8QDz5TRHFqHV52EOFjts/edit#) which discuss this but in short: The pattern was hard to work with because of shared state and led to more data being sent to readers than was needed

### Loadable
Is now gone. We used this for lazy loading react components but this is now acheived via the islands pattern

### `CAPIBrowserType`
Is now gone. This was used type as part of the data set at `window.guardian.app` which is now deleted

### `window.guardian.app`
Is now gone. We stored data for here to later use as part of hydration. The islands pattern however serialises each islands data directly so there's no need for a centralised object on `window`

### `webpack-manifest-plugin`
Added to replace functionality that was previously provided by Loadable

### `GA` > `GAData`
Renamed to be clearer about what this property is

### CODE REVIEW

File: dotcom-rendering/src/model/window-guardian.ts [hunk 1]
```diff
@@ -66,165 +68,9 @@ const makeWindowGuardianConfig = (
...
-		contributionsServiceUrl,
-		isImmersive: CAPI.isImmersive,
-		isPhotoEssay: CAPI.config.isPhotoEssay || false,
-		isSpecialReport: CAPI.isSpecialReport,
-		isLiveBlog: CAPI.config.isLiveBlog || false,
-		isLive: CAPI.config.isLive || false,
-		isPreview: CAPI.config.isPreview,
-		stage: CAPI.config.stage,
-		matchUrl: CAPI.matchUrl,
-		elementsToHydrate: CAPI.blocks
```
Comment (mxdvl) [inline dotcom-rendering/src/model/window-guardian.ts:207]: Have we checked that nothing is expecting this to be present on the page?

I’m thinking mostly of @guardian/commercial-dev

Comment (oliverlloyd) [reply dotcom-rendering/src/model/window-guardian.ts:207]: I pinged comm dev and they said they are only using `window.guardian.config`. Specifically:

```
Ashish Puliyel, Yesterday 16:58
I'm afraid of commitment. So I'm scared to say we never use it. But I've looked and not seen anywhere where we use window.guardian.app
```

Comment (mxdvl) [reply dotcom-rendering/src/model/window-guardian.ts:207]: Oh, right, I didn’t see it was kept on L260.

**Review by mxdvl — APPROVED** [review]
Begone!

File: dotcom-rendering/src/lib/assets.ts [hunk 2]
```diff
@@ -33,40 +32,30 @@ const decideAssetOrigin = (stage: string | undefined): string => {
...
-	// Builds with the SKIP_LEGACY flag will not include legacy bundles, this aids build performance.
-	// 'ophan.87b473fc83e9ca6250fc.js' -> 'ophan'
-	const chunkName = filename.split('.')[0];
-	const chunks: string[] | undefined =
-		loadableManifestLegacy.assetsByChunkName?.[chunkName];
-	const legacyFilename = chunks && chunks.length > 0 && chunks[0];
+	const filename = isDev ? file : manifest[file];
+	const legacyFilename = isDev
+		? file.replace('.js', '.legacy.js')
+		: legacyManifest[file];
```
Comment (oliverlloyd) [inline dotcom-rendering/src/lib/assets.ts:43]: So much clearer! 😻

File: dotcom-rendering/src/web/browser/sentryLoader/sentry.ts [hunk 3]
```diff
@@ -50,11 +51,6 @@ Sentry.init({
 	},
 });
 
-Sentry.configureScope((scope) => {
-	scope.setTag('edition', editionLongForm);
-	scope.setTag('contentType', contentType);
-});
```
Comment (oliverlloyd) [inline dotcom-rendering/src/web/browser/sentryLoader/sentry.ts:56]: Removed in favour of simplicity.

It made sense to include this extra data on errors when it was freely available but now we'd need to refactor the window object to also include these properties which feels excessive

**Review by oliverlloyd — COMMENTED** [review]
I've read through the code and nothing is jumping out to me here. I can't approve because I'm an author but nonetheless I do heartily approve.

**Review by OllysCoding — APPROVED** [review]
So much code removed 😍

**Comment by mxdvl** [comment]
🏝️  #3629

**Pull request merged**